src/test/java/de/tum/in/www1/artemis/AccountResourceIntegrationTest.java

```diff
@@ -50,27 +88,129 @@ public void registerAccount() throws Exception {
         request.postWithoutLocation("/api/register", userVM, HttpStatus.CREATED, null);
     }
 
+    @Test
+    public void registerAccountTooLongPassword() throws Exception {
+        // setup user
+        User user = ModelFactory.generateActivatedUser("ab123cd");
+        ManagedUserVM userVM = new ManagedUserVM(user);
+        userVM.setPassword("e".repeat(ManagedUserVM.PASSWORD_MAX_LENGTH + 1));
+
+        // make request
+        request.postWithoutLocation("/api/register", userVM, HttpStatus.BAD_REQUEST, null);
+    }
+
+    @Test
+    public void registerAccountTooShortPassword() throws Exception {
+        // setup user
+        User user = ModelFactory.generateActivatedUser("ab123cd");
+        ManagedUserVM userVM = new ManagedUserVM(user);
+        userVM.setPassword("e".repeat(Math.max(0, ManagedUserVM.PASSWORD_MIN_LENGTH - 1)));
+
+        // make request
+        request.postWithoutLocation("/api/register", userVM, HttpStatus.BAD_REQUEST, null);
+    }
+
+    @Test
+    public void registerAccountEmptyPassword() throws Exception {
+        // setup user
+        User user = ModelFactory.generateActivatedUser("ab123cd");
+        ManagedUserVM userVM = new ManagedUserVM(user);
+        userVM.setPassword("");
+
+        // make request
+        request.postWithoutLocation("/api/register", userVM, HttpStatus.BAD_REQUEST, null);
+    }
+
+    @Test
+    public void registerAccountRegistrationDisabled() throws Throwable {
+        testWithRegistrationDisabled(() -> {
+            // setup user
+            User user = ModelFactory.generateActivatedUser("ab123cd");
+            ManagedUserVM userVM = new ManagedUserVM(user);
+            userVM.setPassword("password");
+
+            // make request
+            request.postWithoutLocation("/api/register", userVM, HttpStatus.FORBIDDEN, null);
+        });
+    }
+
+    @Test
+    public void registerAccountRegistrationConfigEmpty() throws Throwable {
+        testWithChangedConfig("registrationEnabled", Optional.empty(), () -> {
+            // setup user
+            User user = ModelFactory.generateActivatedUser("ab123cd");
+            ManagedUserVM userVM = new ManagedUserVM(user);
+            userVM.setPassword("password");
+
+            // make request
+            request.postWithoutLocation("/api/register", userVM, HttpStatus.FORBIDDEN, null);
+        });
+    }
+
+    @Test
+    public void registerAccountInvalidEmail() throws Throwable {
+        // Inject email-pattern to be independent of the config
+        testWithChangedConfig("allowedEmailPattern", Optional.of(Pattern.compile("[a-zA-Z0-9_\\-.+]+@[a-zA-Z0-9_\\-.]+\\.[a-zA-Z]{2,5}")), () -> {
```

Comment:
nit: This is not a valid email regex. Eg: `_@-.com` Matching emails is a pain but I think this tests what it's supposed to test.

Reply:
Indeed, the test is just supposed to verify the request gets rejected if the Email does not match; I'm afraid it'll be impossible to verify the correctness of the pattern since its a config that may be changed for different installations and therefore there's no clear requirements it could be tested against.